boot/bootutil/src/loader.c

```diff
@@ -1549,89 +1559,432 @@ boot_set_image_ok(void)
 #endif /* !MCUBOOT_OVERWRITE_ONLY */
 
 /**
- * Performs an image swap if one is required.
+ * Sets the active image on which the bootloader will continue to work.
+ *
+ * @param image_id                  The index of the image (from 0).
+ */
+static void
+boot_set_active_image(size_t image_id)
```

Comment:
This function seems to be always called in the following pattern: `1) FOR EACH IMAGE; 2) SET_ACTIVE; 3) DO SOMETHING`. Maybe (1) and (2) could be a macro which in a single-image does nothing...

Reply:
maybe something like this? 
```
ITERATE_THROUGH_IMAGES(img_cnt) {
....
}
```

Reply:
The reorganization of the code made it unnecessary.